File: ITP1_6_C.py
# -*- coding:utf-8 -*-
arr = [[[0 for i3 in range(10)] for i2 in range(3)] for i1 in range(4)]

n = int(input())
for i in range(n):
    b,f,r,v = map(int,input().split())
    arr[b-1][f-1][r-1] += v

for i in range(4):
    for j in range(3):
        for k in range(10):
            print(' ',arr[i][j][k],sep = '',end='')
        print('')
    if i != 3:
        print('####################')

File: ITP1_6_C.cpp
#include<iostream>
using namespace std;

int main() {
  int n;
  cin >> n;

  int arr[5][4][11];
  for (int i = 0; i < 5; i++) {
    for (int j = 0; j < 4; j++) {
      for (int k = 0; k < 11; k++) {
	arr[i][j][k] = 0;
      }
    }
  }
  
  int b, f, r, v;
  for (int i = 0; i < n; i++) {
    cin >> b >> f >> r >> v;
    arr[b][f][r] += v;
  }

  for (int i = 1; i < 5; i++) {
    for (int j = 1; j < 4; j++) {
      for (int k = 1; k < 11; k++) {
	cout << " "  << arr[i][j][k];
      }
      cout << endl;
    }
    if (i != 4) cout << "####################" << endl;
  }
  
  return 0;
}
